src/test/schema_test.cpp

```diff
@@ -735,3 +735,584 @@ TEST(SchemaTest, ModelConfigNireqNegative) {
+                        ],
+                        "outputs": [
+                            {"data_item": "a",
+                            "alias": "new_dummy_output"}
+                        ]
+                    }
+                ],
+                "outputs": [
+                    {"custom_dummy_output": {"node_name": "dummyNode",
+                                            "data_item": "new_dummy_output"}
+                    }
+                ]
+            }
+        ]
+    })";
+
+    rapidjson::Document customNodeConfigModelNameLibraryNameParsed;
+    customNodeConfigModelNameLibraryNameParsed.Parse(customNodeConfigModelNameLibraryName);
+    auto result = ovms::validateJsonAgainstSchema(customNodeConfigModelNameLibraryNameParsed, ovms::MODELS_CONFIG_SCHEMA);
+    EXPECT_EQ(result, ovms::StatusCode::JSON_INVALID);
+}
+
+TEST(SchemaTest, ModelNodeConfigLibraryName) {
+    const char* modelNodeConfigLibraryName = R"(
+    {
+        "model_config_list": [
+            {
+                "config": {
+                    "name": "dummy",
+                    "base_path": "dummy_path",
+                    "target_device": "CPU",
+                    "model_version_policy": {"all": {}},
+                    "nireq": 1
+                }
+            }
+        ],
+        "custom_node_library_config_list": [
+            {
+                "name": "dummy_library",
+                "base_path": "dummy_path"
+            }
+        ],
+        "pipeline_config_list": [
+            {
+                "name": "pipeline1Dummy",
+                "inputs": ["custom_dummy_input"],
+                "nodes": [
+                    {
+                        "name": "dummyNode",
+                        "library_name": "dummy_library",
+                        "type": "DL model",
+                        "params": {
+                            "a": "1024",
+                            "b": "512"
+                        },
+                        "inputs": [
+                            {"b": {"node_name": "request",
+                                "data_item": "custom_dummy_input"}}
+                        ],
+                        "outputs": [
+                            {"data_item": "a",
+                            "alias": "new_dummy_output"}
+                        ]
+                    }
+                ],
+                "outputs": [
+                    {"custom_dummy_output": {"node_name": "dummyNode",
+                                            "data_item": "new_dummy_output"}
+                    }
+                ]
+            }
+        ]
+    })";
+
+    rapidjson::Document modelNodeConfigLibraryNameParsed;
+    modelNodeConfigLibraryNameParsed.Parse(modelNodeConfigLibraryName);
+    auto result = ovms::validateJsonAgainstSchema(modelNodeConfigLibraryNameParsed, ovms::MODELS_CONFIG_SCHEMA);
+    EXPECT_EQ(result, ovms::StatusCode::JSON_INVALID);
+}
+
+TEST(SchemaTest, ModelNodeConfigModelNameLibraryName) {
```

Comment:
It really should not be a test if library name is allowed in dl node configuration but any not specifically required parameter. For test purpose it would be clearer if you would just name this parameter 
NotAppropiate, DisallowedParameter or anything similar.

Reply:
Similar case with custom node.